src/colvar_neuralnetworkcompute.cpp

```diff
@@ -283,27 +262,1003 @@ std::vector<std::vector<double>> neuralNetworkCompute::multiply_matrix(const std
+        }
+        m_k_weights.push_back(weights_tmp);
+    }
+    for(size_t i=0;i<getD_model();i++)
+    {
+        std::getline(ifs_weights,line);
+        if(startswith(line,"#"))
+        {
+            i--;
+            continue;
+        }
+        std::vector<std::string> splited_data;
+        colvarparse::split_string(line, std::string{" "}, splited_data);
+        
+        std::vector<double> weights_tmp(splited_data.size());
+        for (size_t i = 0; i < splited_data.size(); ++i) 
+        {
+            weights_tmp[i] = std::stod(splited_data[i]);
+        }
+        m_v_weights.push_back(weights_tmp);
+    }
+    for(size_t i=0;i<getD_model();i++)
+    {
+        std::getline(ifs_weights,line);
+        if(startswith(line,"#"))
+        {
+            i--;
+            continue;
+        }
+        std::vector<std::string> splited_data;
+        colvarparse::split_string(line, std::string{" "}, splited_data);
+        
+        std::vector<double> weights_tmp(splited_data.size());
+        for (size_t i = 0; i < splited_data.size(); ++i) 
+        {
+            weights_tmp[i] = std::stod(splited_data[i]);
+        }
+        m_o_weights.push_back(weights_tmp);
+    }
+    // parse biases file
+    std::ifstream ifs_biases(biases_file.c_str());
+    for(size_t i=0;i<getD_model();i++)
+    {
+        std::getline(ifs_biases, line);
+        if(startswith(line,"#"))
+        {
+            i--;
+            continue;
+        }
+        m_q_biases.push_back(std::stod(line));
+    }
+    for(size_t i=0;i<getD_model();i++)
+    {
+        std::getline(ifs_biases, line);
+        if(startswith(line,"#"))
+        {
+            i--;
+            continue;
+        }
+        m_k_biases.push_back(std::stod(line));
+    }
+    for(size_t i=0;i<getD_model();i++)
+    {
+        std::getline(ifs_biases, line);
+        if(startswith(line,"#"))
+        {
+            i--;
+            continue;
+        }
+        m_v_biases.push_back(std::stod(line));
+    }
+    for(size_t i=0;i<getD_model();i++)
+    {
+        std::getline(ifs_biases, line);
+        if(startswith(line,"#"))
+        {
+            i--;
+            continue;
+        }
+        m_o_biases.push_back(std::stod(line));
+    }
+    m_output_size = m_input_size;//why not the end of array m_weights?
+}
+
+// \brief calculate the output value of particular layer, with bias
+std::vector<double> AttentionLayer::compute(const std::vector<double>& input) const {
+    std::vector<std::vector<double>> q=transpose(reshape(input,int(m_d_model),int(m_input_size/m_d_model)));//seq_len*d_model
+    std::vector<std::vector<double>> k=q;//shape=(seq_len_k,d_model)
+    std::vector<std::vector<double>> v=q;//total assignment
+    std::vector<std::vector<double>> buf=creatematrix(q.size(),1,1);//all one matrix
+    std::vector<std::vector<double>> q_expand=hstack(q,buf);
+    std::vector<std::vector<double>> k_expand=q_expand;
+    std::vector<std::vector<double>> v_expand=q_expand;
```

Comment:
These variables are not used in `compute` but only `computeGradient`. If you want to reuse them to save CPU cycles you can make them member variables. Otherwise they should be removed.